packages/autometrics/src/platform.node.ts

```diff
@@ -140,6 +140,7 @@ function readClosest(path: string): Uint8Array {
       return readFileSync(join(basePath, path));
     } catch {
       basePath = dirname(basePath);
+      break;
```

Comment:
so i think we need to modify the `while` condition instead of adding a break here, since the break would prevent us from recursively searching parent directories. 

the infinite loop you encountered seems to happen because `dirname("/") === "/"`, so we'll never reach the breaking case of `basePath.length === 0` in the while condition when we're dealing with full paths

so, we need a base case in the while loop that covers when the basePath is reduced to `"/"`. but also to be fully cross-platform, we should consider what the equivalent windows file path(s) might look like

Reply:
also worth noting is the node.js `dirname` function returns `"."` for the empty string (https://github.com/nodejs/node/blob/0161ad0baf87a0009101ce00b22b874ad6fc5d88/lib/path.js#L660)

and there's some special handling in `dirname` for windows device roots (see [here](https://github.com/nodejs/node/blob/main/lib/path.js#L718))

i might say for now we create a util function `isRootPath` that can also be used in the while loop's condition. 

basically, we need to beak the loop when `dirname` hits a root

Reply:
perfect — that's a good cross-platform solution. and indeed it was better to keep that break inside the catch (instead of the while loop condition), to ensure we also actually check the fs root